src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -2502,10 +2502,10 @@ void InstanceKlass::clean_implementors_list() {
     assert (ClassUnloading, "only called for ClassUnloading");
     for (;;) {
       // Use load_acquire due to competing with inserts
-      InstanceKlass* impl = Atomic::load_acquire(adr_implementor());
+      InstanceKlass* volatile* iklass = adr_implementor();
+      InstanceKlass* impl = (iklass != nullptr) ? Atomic::load_acquire(iklass) : nullptr;
```

Comment:
This looks very klunky as we do a raw read, check it for null then re-read with acquire semantics. Cleaner IMO to do a raw read followed by a raw `OrderAccess::acquire()` and no need for a null check.